.github/workflows/docc.yml

```diff
@@ -0,0 +1,108 @@
+      - name: Extract Swift 5.5.1
+        run: tar xzf swift-5.5.1-RELEASE-ubuntu20.04.tar.gz
+      - name: Add Swift toolchain to PATH
+        run: |
+          echo "$GITHUB_WORKSPACE/swift-5.5.1-RELEASE-ubuntu20.04/usr/bin" >> $GITHUB_PATH
+      - name: Checkout swift-docc
+        uses: actions/checkout@v2
+        with:
+          repository: apple/swift-docc
+          ref: main
+          path: swift-docc
+      - name: Cache DocC
+        id: cache-docc
+        uses: actions/cache@v2
+        with:
+          key: swift-url-docc-build
+          path: swift-docc/.build
+      - name: Build swift-docc
+        if: ${{ !steps.cache-docc.outputs.cache-hit }}
+        run: |
+          cd swift-docc; swift build --product docc -c release; cd ..
+      - name: Checkout swift-docc-render
+        uses: actions/checkout@v2
+        with:
+          repository: apple/swift-docc-render
+          ref: main
+          path: swift-docc-render
+      - name: Build swift-docc-render
+        run: |
+          cd swift-docc-render; npm install && npm run build; cd ..
+      - name: Checkout gh-pages Branch
+        uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+          path: docs-out
+      - name: Build documentation
+        run: >
+          rm -rf docs-out/.git;
+          rm -rf docs-out/main;
+          for tag in $(echo "main"; git tag | tail -n +94);
+          do
+            echo "⏳ Generating documentation for "$tag" release.";
+            
+            if [ -d "docs-out/$tag" ] 
+            then 
+                echo "✅ Documentation for "$tag" already exists.";
+            else 
+                git checkout -f "$tag";
+                mkdir -p Sources/ProjectDescription/Documentation.docc;
+                export DOCC_HTML_DIR="$(pwd)/swift-docc-render/dist";
+                rm -rf .build/symbol-graphs;
+                mkdir -p .build/symbol-graphs;
+                swift build \
+                  --target ProjectDescription \
+                  -Xswiftc \
+                  -emit-symbol-graph \
+                  -Xswiftc \
+                  -emit-symbol-graph-dir \
+                  -Xswiftc \
+                  .build/symbol-graphs \
+                  && swift-docc/.build/release/docc convert Sources/ProjectDescription/Documentation.docc \
+                    --fallback-display-name ProjectDescription \
+                    --fallback-bundle-identifier io.tuist.ProjectDescription \
+                    --fallback-bundle-version 0.0.0 \
+                    --additional-symbol-graph-dir \
+                    .build/symbol-graphs \
+                    --transform-for-static-hosting \
+                    --hosting-base-path /tuist/"$tag" \
+                    --output-path docs-out/"$tag" \
+                    && echo "✅ Documentation generated for "$tag" release." \
+                    || echo "⚠️ Documentation skipped for "$tag".";
```

Comment:
Not sure I got the logic behind `&&`  and `||` here 🤔

Reply:
It's used for success/failure